src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -185,3 +186,232 @@ If you publish your template as *Unlisted*, you can test it under exactly the sa
 . Click *Publish*.
 . When publishing is completed, click the link to view and test your updated template.
 . When you're ready to give general users access, clear *Unlisted* and republish the template.
+
+== Using the Koji debugger
+
+The Koji debugger enables live testing and debugging in a production environment.
+It allows you to follow the communication happening between the Koji platform and the template.
+
+[NOTE]
+The debugger is only available after you publish your template for the first time.
+
+To open the debugger, go to *My repositories* and click *Open debugger* next to the project that you want to test.
+You can also open it by clicking *Open debugger* in the Koji editor.
+The debugger will open in a new browser tab.
+
+=== Debugging contexts
+
+A set of tabs allow you to select the context you want to debug.
+See <<_using_live_previews>> for an explanation of contexts.
+For example, if you select *Remix*, tools for debugging the remix context will be displayed.
+
+Any context can be selected, whether or not it is enabled for your Koji.
+If you select a context that is not enabled, a message will be displayed saying that it is unavailable.
+
+The following contexts are available for selection.
+
+==== Default
+
+Selecting *Default* shows you a preview of the running template.
+Events associated with the default context are displayed in an event pane.
+See <<_viewing_events>>.
+
+In the *Default* context, you can see what happens when the template asks the platform for things like <<withkoji-koji-iap-package#, in-app purchases>> or <<withkoji-koji-auth-sdk#, auth grants>>.
+
+==== Remix
+
+Selecting *Remix* displays a special remixing page, if you defined one in your template.
+Otherwise, it displays the editable elements of your template and allows you to change them.
+A separate pane displays the *VCC Values* as they appear in the code for the template.
+Events associated with the *Remix* context are displayed in an event pane.
+See <<_viewing_events>>.
+
+*Remix* gives you a dynamic view of remix events, for a better understanding of how VCC data is changing over time in response to user actions.
+
+==== Thumbnail
+
+Selecting the *Thumbnail* shows you exactly how your template will look if you share it on social media.
+See <<rich-preview-image#>>.
+
+*Thumbnail* shows previews in both square and rectangular format.
+
+==== Admin
+
+If the admin context is enabled, selecting *Admin* displays the admin page.
```

Comment:
Here's another example of redundant phrasing (admin). One possible fix (possibly not the best):
```suggestion
Select *Admin* to debug the admin context, if you defined one in your template.
```